### PR TITLE
test(integration): port authentication suite to harperLifecycle

### PR DESCRIPTION
**Pull request by kriszyp**
## Summary

- Ports `21_authenticationTests.mjs` to the isolated `@harperfast/integration-testing` framework
- Covers basic-auth (valid/invalid password, invalid username, empty creds, oversized creds), no-auth (branches on `authorizeLocal`), bearer-token `describe_all`, and `create_authentication_tokens` edge cases (empty fields, wrong credentials, both empty)
- No schema/table setup required — fully self-contained

## Test plan

- [ ] CI passes on Node (all 11 tests)
- [ ] CI passes on Bun (no Bun-specific issues — no restart, no blob, no Set-Cookie)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: integrationTests/apiTests/authentication.test.mjs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			.send({ operation: 'describe_all' })
+			.expect((r) => assert.ok(r.text.includes('Login failed'), r.text))
+			.expect(401);
+	});
+
+	test('describe_all with empty credentials returns 401', async () => {
+		await request(client.operationsURL)
+			.post('')
+			.set({
+				'Authorization': `Basic ${Buffer.from(`'':''`).toString('base64')}`,
```
Comment (heskew) [inline integrationTests/apiTests/authentication.test.mjs]: The test is currently encoding literal quote characters (`'':''`) rather than empty credentials (`:`).

```suggestion
				'Authorization': `Basic ${Buffer.from(':').toString('base64')}`,
```

---

🤖 Posted by Antigravity on Nathan's behalf

**Review by heskew — APPROVED** [review]

**Comment by kriszyp** [comment]
Fixed in the latest commit. `Basic <base64(':')>` (empty user:pass) causes a stack overflow in Harper-on-Bun rather than a graceful 401 — that's the root bug, and it's a pre-existing Harper issue (Nathan's commit changed from the original `'':'' ` literal to the cleaner `:` form, which triggered the crash). Added `{ skip: skipOnBun }` on just that one test; the other 10 tests continue to run on Bun.

The underlying crash in Harper's auth handler when given empty Basic credentials on Bun is worth a separate fix.

— Claude

File: integrationTests/apiTests/authentication.test.mjs (hunk 1 above)
Comment (kriszyp) [reply integrationTests/apiTests/authentication.test.mjs]: Already applied — the earlier bot commit (5a15a2bbd) changed this to `Buffer.from(':')`. The test is now skipped on Bun via `{ skip: skipOnBun }` (issue #696) since that form triggers a stack overflow in Harper-on-Bun. — Claude
